### PR TITLE
storage/lock_manager: Add metrics to the new lock waiting queue

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What is changed and how it works?

Issue Number: Ref #13298

What's Changed:

```commit-message
Add metrics to the new lock waiting queue, including:

* The number of keys on which there is lock waiting
* The number of requests that are waiting in the queue
* The histogram of the queue length observed when enqueueing new requests
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (WIP)

Side effects

- Performance regression
    - Consumes more CPU

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/storage/metrics.rs [hunk 1]
```diff
@@ -575,4 +584,19 @@ lazy_static! {
...
+        LockWaitQueueEntriesGauge,
+        "tikv_lock_wait_queue_entries_gauge_vec",
+        "Statistics of the lock wait queue's state",
+        &["type"]
+    )
+    .unwrap();
+
+    pub static ref LOCK_WAIT_QUEUE_LENGTH_HISTOGRAM: Histogram = register_histogram!(
+        "tikv_lock_wait_queue_length",
+        "Statics of length of queues counted when enqueueing",
```
Comment (TonsnakeLin) [inline src/storage/metrics.rs:598]:
```suggestion
        "Statistics  of length of queues counted when enqueueing",
```

**Review by sticnarf — APPROVED** [review]

**Review by ekexium — APPROVED** [review]

**Comment by ekexium** [comment]
Shall we benchmark it?

**Comment by cfzjywxk** [comment]
> Shall we benchmark it?

We could do the benchmark in the enabling PR later.

**Pull request merged**